### PR TITLE
DOCS: add modal and attachment option docs, and refactor some docs

### PR DESCRIPTION
**Pull request by Kigstn**
## What type of pull request is this?

- [ ] Non-breaking code change
- [ ] Breaking code change
- [x] Documentation change/addition

## Description

This adds docs for modals and attachment option and reactors some outdated info

## Changes

- add modal guide
- add attachment option to command guide
- change `target_id` to `target` in context menu guide
- add additional `@listen()` styles to event docs
- update the guide indexing. this should help prevent that in the future

## Checklist

- [ ] I've formatted my code with [Black](https://black.readthedocs.io/en/stable/)
- [ ] I've ensured my code works on `Python 3.10.x`
- [ ] I've tested my code

### CODE REVIEW

**Comment by Kigstn** [comment]
Note: This depends on pr #265, #186, #266

**Comment by LordOfPolls** [comment]
image
Suggest using newer screenshots, this bug with modals has been fixed

File: docs/src/Guides/01 Getting Started.md [hunk 1]
```diff
@@ -79,4 +79,4 @@ Congratulations! You now have a basic understanding of this library.
 If you have any questions check out our other guides, or join the
 --8<-- "discord_inv.md"
 
-For more examples, check out the [examples page](/Guides/99 Example)
+For more examples, check out the [examples page](/Guides/90 Example)
```
Comment (leestarb) [inline docs/src/Guides/01 Getting Started.md:82]: The examples url is now invalid
```suggestion
For more examples, check out the [examples page](/Guides/99 Example)
```

**Review by leestarb — CHANGES_REQUESTED** [review]

File: docs/src/Guides/01 Getting Started.md (hunk 1 above)
Comment (Kigstn) [reply docs/src/Guides/01 Getting Started.md:82]: Good catch, forgot to rename that

Comment (leestarb) [reply docs/src/Guides/01 Getting Started.md:82]: Why did you rename it?

**Review by leestarb — APPROVED** [review]
Looks good

**Review by silasary — APPROVED** [review]

**Pull request merged**